src/MatrixEventHandler.ts

```diff
@@ -156,22 +156,48 @@ export class MatrixEventHandler {
                 }
             } else if (event.content.is_direct && bridgeBot.isRemoteUser(event.state_key)) {
                 log.debug("Got request to PM", event.state_key);
-                const {
-                    username,
-                    protocol,
-                } = this.purple.getUsernameFromMxid(event.state_key!, this.config.bridge.userPrefix);
-                log.debug("Mapped username to", username);
-                const remoteData = {
+                let remoteData = {
                     matrixUser: event.sender,
-                    protocol_id: protocol.id,
-                    recipient: username,
                 } as any;
+
+                // this is not always possible (could be a brand new user being reached),
+                // but is preferable to decoding the mxid because of potential information loss
+                // during mxid generation (see GH-268)
+                const remoteUsers = await this.store.getRemoteUsersFromMxId(event.state_key!);
+                if (remoteUsers.length > 1) {
+                    log.error(
+                        `Multiple remote users found for ${event.state_key!}:`,
+                        remoteUsers.map(u => `${u.protocolId}://${u.username}`).join(', ')
```

Comment:
Is this supposed to look like an URI with `://`?
The lines below just separate the protocol and username with a colon.

Reply:
Yeah – it follows the convention we often use in the logs, like [here](https://github.com/matrix-org/matrix-bifrost/blob/develop/src/Program.ts#L300).

The regular colon usage below is mostly arbitrary, I think. @Half-Shot?